shell/platform/darwin/ios/framework/Source/FlutterPlatformViews.mm

```diff
@@ -593,7 +596,13 @@ - (void)blockGesture {
       // FlutterPlatformViewGestureRecognizersBlockingPolicyEager, but we should try it if a similar
       // issue arises for the other policy.
       if (@available(iOS 18.2, *)) {
-        if ([self containsWebView:self.embeddedView]) {
+        // This workaround is designed for WKWebView only. The 1P web view plugin provides a
+        // WKWebView itself as the platform view. However, some 3P plugins provide wrappers of
+        // WKWebView instead. So we perform DFS to search the view hierarchy (with a depth limit).
+        // Passing a limit of 0 means only searching for platform view itself; Pass 1 to include its
+        // children as well, and so on. We should be conservative and start with a small number. The
+        // AdMob banner has a WKWebView at depth 7.
+        if ([self containsWebView:self.embeddedView remainingSubviewDepth:1]) {
```

Comment:
Passing 1 means searching for its children only. AdMob is at depth 7. I think we should start small here to be conservative.